### PR TITLE
[IMP] point_of_sale, pos_*: use x_commands from frontend.

### PR DESCRIPTION
**Pull request by davidmonnom**
*: pos_sale, pos_loyalty, pos_restaurant, pos_event

Before all lines was sent to the backend to be processed, if the backend
had a line that wasn't sent by the frontend, it was removed from the
order.

Now the frontend sends only modified lines to the backend, if a line is
deleted from the frontend, it will be deleted from the backend too. But
the backend doesn't remove lines that weren't sent by the frontend.

This change was made to avoid losing information when the backend delete
a line that wasn't sent by the frontend.

### CODE REVIEW

File: addons/point_of_sale/static/src/app/store/pos_store.js [hunk 1]
```diff
@@ -964,6 +971,9 @@ export class PosStore extends Reactive {
             this.preSyncAllOrders(orders);
             const context = this.getSyncAllOrdersContext(orders);
 
+            if (this.pendingOrder.delete.size) {
```
Comment (adgu-odoo) [inline addons/point_of_sale/static/src/app/store/pos_store.js:974]: Why not add use the changes made to `getPendingOrder`. You retrieve `orderToDelete` from `this.getPendingOrder()` and do something like `await this.deleteOrders([], orderToDelete.map((o) => o.id));`

File: addons/pos_loyalty/static/src/overrides/models/pos_store.js [hunk 2]
```diff
@@ -563,7 +563,10 @@ patch(PosStore.prototype, {
```
Comment (adgu-odoo) [inline addons/pos_loyalty/static/src/overrides/models/pos_store.js:563]: This method signature is `async processServerData(loadedData)` but the parent does not have parameters, as you do not use the parameter here you can remove it on the signature and make it `async processServerData()`

File: addons/pos_sale/static/src/overrides/models/pos_store.js [hunk 3]
```diff
@@ -150,12 +150,12 @@ patch(PosStore.prototype, {
             const product_unit = line.product_id.uom_id;
             if (product_unit && !product_unit.is_pos_groupable) {
                 let remaining_quantity = newLine.qty;
+                newLine.delete();
```
Comment (adgu-odoo) [inline addons/pos_sale/static/src/overrides/models/pos_store.js:153]: Why this change ?

**Review by adgu-odoo — CHANGES_REQUESTED** [review]

File: addons/pos_sale/static/src/overrides/models/pos_store.js (hunk 3 above)
Comment (davidmonnom) [reply addons/pos_sale/static/src/overrides/models/pos_store.js:153]: Otherwise there is a bug with remaining_quantity, that lead to creation of useless lines

**Comment by adgu-odoo** [comment]
@robodoo r+
